Wrappers/Python/test/test_DataContainer.py

```diff
@@ -467,6 +467,41 @@ def test_ImageData(self):
         self.assertNumpyArrayEqual(numpy.asarray(data.shape), data.as_array().shape)
 
 
+    def test_ImageData_from_numpy(self):
+        """
+        Test the creation and manipulation of ImageData from a numpy array.
+        """
+        # Initialize array and ImageGeometry
+        arr = numpy.arange(24, dtype=numpy.float32).reshape(2, 3, 4)
+        ig = ImageGeometry(voxel_num_x=4, voxel_num_y=3, voxel_num_z=2)
+        data = ImageData(arr, deep_copy=True, geometry=ig)
+        
+        # Check initial shape, geometry, and deep copy
+        self.assertEqual(data.shape, (2, 3, 4))
+        self.assertEqual(data.geometry, ig)
+        self.assertNotEqual(id(data.array), id(arr))
+    
+        # Fill with zeros and check
+        data.fill(0)
+        numpy.testing.assert_array_equal(data.as_array(), numpy.zeros(ig.shape))
```

Comment:
do we get the same type by default? Or maybe checking equality with 0 in `assert_array_equal` returns `True` comparing different types (int or floats)